part.go

```diff
@@ -334,18 +335,18 @@ func parseParts(parent *Part, reader *bufio.Reader) error {
 		// Look for part header.
 		bbr := bufio.NewReader(br)
 		err = p.setupHeaders(bbr, "")
-		if err == errEmptyHeaderBlock {
+		if errors.Cause(err) == errEmptyHeaderBlock {
 			// Empty header probably means the part didn't use the correct trailing "--" syntax to
 			// close its boundary.
 			if _, err = br.Next(); err != nil {
-				if err == io.EOF || strings.HasSuffix(err.Error(), "EOF") {
+				if errors.Cause(err) == io.EOF || strings.HasSuffix(err.Error(), "EOF") {
 					// There are no more Parts. The error must belong to the parent, because this
 					// part doesn't exist.
 					parent.addWarning(ErrorMissingBoundary, "Boundary %q was not closed correctly",
 						parent.Boundary)
 					break
 				}
-				return fmt.Errorf("error at boundary %v: %v", parent.Boundary, err)
+				return errors.WithMessage(err, fmt.Sprintf("error at boundary %v", parent.Boundary))
```

Comment:
```suggestion
				return errors.Wrapf(err, "error at boundary %v", parent.Boundary)
```

Reply:
Added a comment explaing why `errors.WithMessage` is used there.

Reply:
Makes sense, we should probably at use WithMessagef then.

Reply:
ah, see the comment now, nvm.